code/datums/components/hat_stabilizer.dm

```diff
@@ -107,6 +107,7 @@
 		return
 	if(attached_hat)
 		var/mutable_appearance/worn_overlay = attached_hat.build_worn_icon(default_layer = ABOVE_BODY_FRONT_HEAD_LAYER-0.1, default_icon_file = 'icons/mob/clothing/head/default.dmi')
+		worn_overlay.appearance_flags |= RESET_COLOR
```

Comment:
```suggestion
		worn_overlay.appearance_flags |= RESET_COLOR|KEEP_APART
```
Otherwise coloring the wig/whatever helmet the component is on will also color the hat. Just make sure that the visuals don't break from adding that.

Reply:
this will make the hat render apart from the human, which is likely to cause layering fuckery.
ideal solution IMO would be drawing the hat not to the hair but "next" to it, on the human itself.

Reply:
> this will make the hat render apart from the human, which is likely to cause layering fuckery. ideal solution IMO would be drawing the hat not to the hair but "next" to it, on the human itself.

Need #89702 to do this sanely, I recommend putting this PR on hold for now until rendering methods from said PR are through and then we can render the hat as a "separated worn overlay" from that PR

Reply:
What if we just add reset color now and do it the separated way later

Reply:
> What if we just add reset color now and do it the separated way later

Sure? That's a half fix but I guess I can bundle that with my PR instead